src/s3ql/backends/backblaze.py

```diff
@@ -0,0 +1,704 @@
+
+        key = self._encode_key(key)
+        headers = CaseInsensitiveDict()
+        headers['Range'] = "bytes=0-1"  # Only get first byte
+
+        resp, data = self._do_request('GET',
+                                      '/file/%s/%s%s' % (self.bucket_name, self.prefix, key),
+                                      self.conn_download, headers=headers)
+
+        meta = self._extractmeta(resp, key)
+        return meta
+
+    @retry
+    def get_size(self, key):
+        """Return size of object stored under *key*"""
+        log.debug('started with %s', key)
+
+        key = self._encode_key(key)
+        key_with_prefix = "%s%s" % (self.prefix, key)
+        request_dict = {'bucketId': self.bucket_id,
+                        'startFileName': key_with_prefix,
+                        'maxFileCount': 1}
+
+        body = bytes(json.dumps(request_dict), encoding='UTF-8')
+        resp, body = self._do_request('POST',
+                                      '/b2api/v1/b2_list_file_names',
+                                      self.conn_api,
+                                      body=body)
+
+        j = json.loads(str(body, encoding='UTF-8'))
+        if j['files'][0]['fileName'] == key_with_prefix:
+            return j['files'][0]['contentLength']
+
+        raise NoSuchObject(key_with_prefix)
+
+    @retry
+    @copy_ancestor_docstring
+    def open_read(self, key):
+        log.debug('started with %s', key)
+
+        key = self._encode_key(key)
+
+        resp, data = self._do_request('GET',
+                                      '/file/%s/%s%s' % (self.bucket_name, self.prefix, key),
+                                      self.conn_download, download_body=False)
+
+        meta = self._extractmeta(resp, key)
+
+        return ObjectR(key, self.conn_download, resp, metadata=meta)
+
+    @copy_ancestor_docstring
+    def open_write(self, key, metadata=None, is_compressed=False):
+        log.debug('started with %s', key)
+
+        key = self._encode_key(key)
+
+        return ObjectW(key, self, metadata)
+
+    @retry
+    @copy_ancestor_docstring
+    def clear(self):
+        log.debug('started')
+        for file in self.list():
+            self.delete(file,force=True)
+
+    @retry
+    @copy_ancestor_docstring
+    def delete(self, key, force=False):
+        log.debug('started with %s', key)
+
+        todel_id = self._list_file_version(key)
+
+        if not todel_id and not force:
+            raise NoSuchObject(key)
+
+        self._delete_file_id(todel_id)
+
+    @copy_ancestor_docstring
+    def list(self, prefix=''):
+        log.debug('started with %s', prefix)
+
+        prefix = self._encode_key(prefix)
+
+        next_filename = self.prefix + prefix
+        keys_remaining = True
+
+        while keys_remaining and next_filename is not None:
+            next_filename, filelist = self._list_file_name(next_filename)
+
+            while filelist:
+                file = filelist.pop(0)
```

Comment:
should be: `for file_ in filelist` to not uselessly modify the list and avoid overwriting builtins